### PR TITLE
[item-collection] Fix list role being missing

### LINKED ISSUE #364: role="listbox" attribute missing from <ul> within terra-item-collection component
# Bug Report

## Description
[Originally logged in terra-core](https://github.com/cerner/terra-core/issues/1656) by @juliacalandro 

When we upgraded to version 2.7.0 of terra-list, we received the following accessibility violation from our wdio tests:

_"description": "Ensures elements with an ARIA role that require parent roles are contained by them",
  "help": "Certain ARIA roles must be contained by particular parents",
  "helpUrl": "https://dequeuniversity.com/rules/axe/2.6/aria-required-parent?application=axeAPI",
  "id": "aria-required-parent",
..................
"nodes": [
    {
      "all": [],
      "any": [
        {
          "data": [
            "listbox"
          ],
          "id": "aria-required-parent",
          "impact": "critical",
          "message": "Required ARIA parent role not present: listbox",
          "relatedNodes": []
        }
      ],
      "failureSummary": "Fix any of the following:\n  Required ARIA parent role not present: listbox"_

After inspecting the terra-list code, we noticed that the `role="option"` attribute was getting set on `<li>` within the ListItem component, but a role attribute was not getting set on `<ul>` within the List component. I looked into this issue a bit further, and found [this webpage](http://w3c.github.io/aria-practices/examples/listbox/listbox-scrollable.html) about aria/accessibility which states that `role="listbox"` needs to be an attribute on `<ul>` which is the parent of `<li>`.

## Steps to Reproduce
Install the aXe extension for Chrome.
Run a project using terra-list so that it can be opened and inspected in the browser.
Inspect the page and navigate to where the List component is being used.
Verify that `<ul>` does not have a role set but its child `<li>`s do.
Navigate to the aXe tab and run an inspection on the page.
Verify that an accessibility error is thrown with a description matching the error description above.
Add `role="listbox"` as an attribute to each offending `<ul>` component and run the aXe inspection again.
Verify that the accessibility issue doesn't exist anymore.

## Additional Context / Screenshots
screen shot 2018-06-27 at 11 22 05 am

## Expected Behavior
I expect the accessibility issue mentioned above to no longer be thrown by the terra-list component.

## Possible Solution
Add missing `role="listbox"` attribute to `<ul>` within terra-list component.

## Environment
terra-list 2.6.0
Google Chrome Version 67.0.3396.99 (Official Build) (64-bit)

### PR DESCRIPTION
**Pull request by Blackop778**
### Summary
Item collection lists will have a role tag applied if any of it's children have a chevron (which is the ListItem logic for adding the option role that was causing the issues).

Also adds accessibility testing for the item collection. The found accessibility issue is listed at #378.

### Additional Details
Fixes #364

Thanks for contributing to Terra. 
@cerner/terra

### CODE REVIEW

File: packages/terra-clinical-item-collection/src/_ListView.jsx [hunk 1]
```diff
@@ -39,10 +39,13 @@ function createListItems(children, onSelect, requiredElements) {
 const ListView = ({
   children, onSelect, isListDivided, requiredElements,
 }) => {
+  const hasAChevron = React.Children.toArray(children).some(child => child.props.showListItemChevron);
+
   const listItems = createListItems(children, onSelect, requiredElements);
+  const roleSpread = hasAChevron ? { role: 'listbox' } : {};
```
Comment (emilyrohrbough) [inline packages/terra-clinical-item-collection/src/_ListView.jsx]: Why are we only applying this role when `hasAChevron` is true?  Should we not apply this at all times?

Comment (Blackop778) [reply packages/terra-clinical-item-collection/src/_ListView.jsx]: If the list has `role="listbox"` but doesn't have a child with `role="option"` then that is considered an accessibility violation. `hasAChevron` will be true only if at least one child of the list has `role="option"`.

Comment (emilyrohrbough) [reply packages/terra-clinical-item-collection/src/_ListView.jsx]: I'm not super familiar with list accessibility rules, so should we be adding `role="option"` to each list item regardless of the chevron?

Comment (Blackop778) [reply packages/terra-clinical-item-collection/src/_ListView.jsx]: According to the [specification for the option role](https://www.w3.org/TR/2010/WD-wai-aria-20100916/roles#option) it is for
> A selectable item in a select list.

so I suppose we should be checking for the `isSelectable` prop here and in `terra-list` instead.

File: packages/terra-clinical-item-collection/src/_ListView.jsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ function createListItems(children, onSelect, requiredElements) {
 const ListView = ({
   children, onSelect, isListDivided, requiredElements,
 }) => {
-  const hasAChevron = React.Children.toArray(children).some(child => child.props.showListItemChevron);
+  const hasAChevron = React.Children.toArray(children).some(child => child.props.isSelectable);
```
Comment (emilyrohrbough) [inline packages/terra-clinical-item-collection/src/_ListView.jsx]: Can we update this const to reflect we are checking for isSelectable to toggle the role assigned below?

File: packages/terra-clinical-item-collection/src/_ListView.jsx [hunk 3]
```diff
@@ -39,10 +39,13 @@ function createListItems(children, onSelect, requiredElements) {
...
   children, onSelect, isListDivided, requiredElements,
 }) => {
+  const hasSelectableChild = React.Children.toArray(children).some(child => child.props.isSelectable);
+
   const listItems = createListItems(children, onSelect, requiredElements);
+  const roleSpread = hasSelectableChild ? { role: 'listbox' } : {};
 
   return (
-    <List data-terra-clinical-item-collection-list-view isDivided={isListDivided} >
+    <List data-terra-clinical-item-collection-list-view isDivided={isListDivided} {...roleSpread}>
```
[the remark this answers is not in the document]
Comment (Blackop778) [reply packages/terra-clinical-item-collection/src/_ListView.jsx]: Jest only snapshots the React elements and their props that would be rendered so it wouldn't have noticed the attributes on the HTML element changing.

**Pull request merged**